sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java

```diff
@@ -0,0 +1,350 @@
+        configuration = cfg;
+        logger = LoggerFactory.getLogger(this.getClass());
+
+        createDatabaseAndContainers(configuration);
+
+        partitionKey = containers.get(0).read().block().getProperties().getPartitionKeyDefinition()
+            .getPaths().iterator().next().split("/")[1];
+
+        concurrencyControlSemaphore = new Semaphore(cfg.getConcurrency());
+
+        logger.info("PRE-populating {} documents ....", cfg.getNumberOfPreCreatedDocuments());
+        dataFieldValue = RandomStringUtils.randomAlphabetic(configuration.getDocumentDataFieldSize());
+        createPrePopulatedDocs(configuration.getNumberOfPreCreatedDocuments());
+
+        if (configuration.isEnableJvmStats()) {
+            metricsRegistry.register("gc", new GarbageCollectorMetricSet());
+            metricsRegistry.register("threads", new CachedThreadStatesGaugeSet(10, TimeUnit.SECONDS));
+            metricsRegistry.register("memory", new MemoryUsageGaugeSet());
+        }
+
+        initializeReporter(cfg);
+
+        MeterRegistry registry = configuration.getAzureMonitorMeterRegistry();
+
+        if (registry != null) {
+            BridgeInternal.monitorTelemetry(registry);
+        }
+
+        registry = configuration.getGraphiteMeterRegistry();
+
+        if (registry != null) {
+            BridgeInternal.monitorTelemetry(registry);
+        }
+        prefixUuidForCreate = UUID.randomUUID().toString();
+        random = new Random();
+        parsedReadWriteQueryPct(configuration.getReadWriteQueryPct());
+    }
+
+    public void shutdown() {
+        if (this.databaseCreated) {
+            cosmosAsyncDatabase.delete().block();
+            logger.info("Deleted temporary database {} created for this test", this.configuration.getDatabaseId());
+        } else if (containerToClearAfterTest.size() > 0) {
+            for (String id : containerToClearAfterTest) {
+                cosmosAsyncDatabase.getContainer(id).delete().block();
+                logger.info("Deleted temporary collection {} created for this test", id);
+            }
+        }
+        cosmosClient.close();
+    }
+
+    private void performWorkload(BaseSubscriber<Object> documentSubscriber, OperationType type, long i) throws Exception {
+        Flux<? extends Object> obs;
+        CosmosAsyncContainer container = containers.get((int) i % containers.size());
+        if (type.equals(OperationType.Create)) {
+            PojoizedJson data = BenchmarkHelper.generateDocument(prefixUuidForCreate + i, dataFieldValue, partitionKey,
+                configuration.getDocumentDataFieldCount());
+            obs = container.createItem(data).flux();
+        } else if (type.equals(OperationType.Query)) {
+            CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+            String sqlQuery = "Select top 100 * from c order by c._ts";
+            obs = container.queryItems(sqlQuery, options, PojoizedJson.class).byPage(10);
```

Comment:
so on query, we just run an orderby by query and we expect it to not fail, but we don't validate the result.

am I right?

Reply:
Yes we are not validating result , as it would put extra perf to our sdk query end to end result. We are only expecting it not to fail for a successful result.